shared/src/data/GoodNumber.test.ts

```diff
@@ -0,0 +1,349 @@
+    });
+    it('should return a human readable string', () => {
+      expect(GN.fromDecimalString('1', 18).toString(GNFormat.LOSSY_HUMAN)).toEqual('1.0');
+      expect(GN.fromDecimalString('1000', 18).toString(GNFormat.LOSSY_HUMAN)).toEqual('1,000');
+      expect(GN.fromDecimalString('1000000', 18).toString(GNFormat.LOSSY_HUMAN)).toEqual('1,000,000');
+      expect(GN.fromDecimalString('0.00203982', 18).toString(GNFormat.LOSSY_HUMAN)).toEqual('0.00204');
+      expect(GN.fromDecimalString('0.000000000000000001', 18).toString(GNFormat.LOSSY_HUMAN)).toEqual('1.0E-18');
+    });
+    it('should return a compact human readable string', () => {
+      expect(GN.fromDecimalString('1', 18).toString(GNFormat.LOSSY_HUMAN_COMPACT)).toEqual('1.0');
+      expect(GN.fromDecimalString('1000', 18).toString(GNFormat.LOSSY_HUMAN_COMPACT)).toEqual('1,000');
+      expect(GN.fromDecimalString('1000000', 18).toString(GNFormat.LOSSY_HUMAN_COMPACT)).toEqual('1,000,000');
+      expect(GN.fromDecimalString('1000000000', 18).toString(GNFormat.LOSSY_HUMAN_COMPACT)).toEqual('1.0B');
+      expect(GN.fromDecimalString('1000000000000', 18).toString(GNFormat.LOSSY_HUMAN_COMPACT)).toEqual('1.0T');
+      expect(GN.fromDecimalString('1000000000000000', 18).toString(GNFormat.LOSSY_HUMAN_COMPACT)).toEqual('1000T');
+      expect(GN.fromDecimalString('0.00203982', 18).toString(GNFormat.LOSSY_HUMAN_COMPACT)).toEqual('2.04E-3');
+      expect(GN.fromDecimalString('0.000000000000000001', 18).toString(GNFormat.LOSSY_HUMAN_COMPACT)).toEqual('1E-18');
+    });
+  });
+
+  describe('toBigNumber', () => {
+    it('should return an equivalent BigNumber', () => {
+      expect(GN.fromDecimalString('1', 18).toBigNumber()).toEqual(BigNumber.from('1000000000000000000'));
+    });
+  });
+
+  describe('toJSBI', () => {
+    it('should return an equivalent JSBI', () => {
+      expect(GN.fromDecimalString('1', 18).toJSBI()).toEqual(JSBI.BigInt('1000000000000000000'));
+    });
+  });
+
+  describe('toNumber', () => {
+    it('should return an equivalent number', () => {
+      expect(GN.fromDecimalString('0.0000001', 18).toNumber()).toEqual(0.0000001);
+    });
+  });
+
+  describe('setResolution', () => {
+    it('should set the resolution', () => {
+      expect(GN.fromDecimalString('1', 18).setResolution(9).toString(GNFormat.DECIMAL)).toEqual('1000000000');
+      expect(GN.fromDecimalString('1', 18).setResolution(4).toString(GNFormat.DECIMAL)).toEqual('100000000000000');
+      expect(GN.fromDecimalString('1', 18).setResolution(19).toString(GNFormat.DECIMAL)).toEqual('0.1');
+    });
+  });
+
+  describe('zero', () => {
+    it('should return a zero value', () => {
+      expect(GN.zero(18).toString(GNFormat.DECIMAL)).toEqual('0');
+    });
+  });
+
+  describe('fromDecimalBig', () => {
+    it('should return a decimal string', () => {
+      expect(GN.fromDecimalBig(new Big('1'), 18).toString(GNFormat.DECIMAL)).toEqual('1');
```

Comment:
I'd like to see more cases here. For example, what happens if the BigSource is `'0.1'`?

Reply:
Will add follow-up PR(s) with more tests. This is just to get us started.